test/24_device_additions.ts

```diff
@@ -89,65 +110,81 @@ export default () => {
+						'deviceInPostProvisioningState',
+						'post-provisioning',
+					);
 				});
 
 				describe('Given a device disconnected from the vpn', () => {
-					it('should have an Idle overall_status when the heartbeat is Online', async () => {
-						await testOverallStatus(
-							ctx.user,
-							ctx.deviceVpnOffHeartbeatOnline.id,
-							'idle',
-						);
-					});
-
-					it('should have an Idle overall_status when the heartbeat is Timeout', async () => {
-						await testOverallStatus(
-							ctx.user,
-							ctx.deviceVpnOffHeartbeatTimeout.id,
-							'idle',
-						);
-					});
-
-					it('should have an Offline overall_status when the heartbeat is Offline', async () => {
-						await testOverallStatus(
-							ctx.user,
-							ctx.deviceVpnOffHeartbeatOffline.id,
-							'offline',
-						);
-					});
+					itShouldHaveOverallStatus(
+						'should have a ${OVERALL_STATUS} overall_status when the heartbeat is Online',
+						'deviceVpnOffHeartbeatOnline',
+						versions.gt(version, 'v6') ? 'reduced-functionality' : 'idle',
+					);
+
+					itShouldHaveOverallStatus(
+						'should have a ${OVERALL_STATUS} overall_status when the heartbeat is Timeout',
+						'deviceVpnOffHeartbeatTimeout',
+						versions.gt(version, 'v6') ? 'reduced-functionality' : 'idle',
+					);
+
+					itShouldHaveOverallStatus(
+						'should have an ${OVERALL_STATUS} overall_status when the heartbeat is Online and RESIN_SUPERVISOR_VPN_CONTROL is false',
+						'deviceWithDeviceConfigVPNDisabled',
+						versions.gt(version, 'v6') ? 'operational' : 'idle',
+					);
+
+					itShouldHaveOverallStatus(
+						'should have an ${OVERALL_STATUS} overall_status when the heartbeat is Online and RESIN_SUPERVISOR_VPN_CONTROL is true',
+						'deviceWithDeviceConfigVPNEnabled',
+						versions.gt(version, 'v6') ? 'reduced-functionality' : 'idle',
+					);
+
+					itShouldHaveOverallStatus(
+						'should have an ${OVERALL_STATUS} overall_status when the heartbeat is Online and RESIN_SUPERVISOR_VPN_CONTROL is false in application config variables',
+						'deviceWithApplicationConfigVPNDisabled',
+						versions.gt(version, 'v6') ? 'operational' : 'idle',
+					);
+
+					itShouldHaveOverallStatus(
+						'should have a ${OVERALL_STATUS} overall_status when the heartbeat is Offline',
+						'deviceVpnOffHeartbeatOffline',
+						versions.gt(version, 'v6') ? 'disconnected' : 'offline',
+					);
+
+					itShouldHaveOverallStatus(
+						'should have a ${OVERALL_STATUS} overall_status when heartbeat is unknown and last connectivity event does not exist',
+						'deviceInPreProvisioningState',
+						'configuring',
+					);
 				});
 
 				describe('Given a device downloading a multicontainer release', () => {
-					it('should properly calculate the overall_status', async () => {
-						await testOverallStatus(
-							ctx.user,
-							ctx.deviceUpdating.id,
-							'updating',
-						);
-					});
+					itShouldHaveOverallStatus(
```

Comment:
Let's also add 2 more test cases for an updating container while vpn is off and while heartbeat is off & test overall_progress for them as well.